pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -326,11 +326,31 @@ func (c *Cluster) postReconcileUpdateOSDProperties(desiredOSDs map[int]*OSDInfo)
 	}
 	logger.Debugf("post processing osd properties with %d actual osds from ceph osd df and %d existing osds found during reconcile", len(osdUsage.OSDNodes), len(desiredOSDs))
 	for _, actualOSD := range osdUsage.OSDNodes {
-		if desiredOSD, ok := desiredOSDs[actualOSD.ID]; ok {
-			if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
-				// Log the error and allow other updates to continue
-				logger.Error(err)
-			}
+		desiredOSD, ok := desiredOSDs[actualOSD.ID]
+		if !ok {
+			return nil
+		}
+		if err := c.resizeOsdCrushWeight(actualOSD); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(err)
+		}
+		if err := c.updateDeviceClassIfChanged(actualOSD.ID, desiredOSD.DeviceClass, actualOSD.DeviceClass); err != nil {
+			// Log the error and allow other updates to continue
+			logger.Error(err)
+		}
+	}
+	return nil
+}
+
+func (c *Cluster) resizeOsdCrushWeight(actualOSD cephclient.OSDNodeUsage) error {
+	if !c.spec.Storage.ResizeOsdCrushWeight {
+		return nil
+	}
+	if actualOSD.KB != actualOSD.CrushWeight {
+		args := []string{"osd", "crush", "reweight", fmt.Sprintf("osd.%d", actualOSD.ID), fmt.Sprintf("%q", actualOSD.KB.String())}
+		buf, err := cephclient.NewCephCommand(c.context, c.clusterInfo, args).Run()
+		if err != nil {
+			return errors.Wrap(err, "failed to reweight. "+string(buf))
```

Comment:
`errors.Wrap` uses a colon to separate error messages. Let's also capture the OSD ID in the error message to help debug. 

```suggestion
			return errors.Wrapf(err, "failed to reweight osd %d: %s", actualOSD.ID, string(buf))
```